### ISSUE TITLE
Event details design improvements

### ISSUE THREAD

**Issue by ChristophWurst** [nextcloud/calendar#3543]
Right now event details are filled in and edited via a tabbed sidebar. We could move this view into a modal and benefit from more vertical space. That space will allow us to render the free/busy view right away right below attendees and rooms.

Design-wise this modal could looks similar to the setting modals. What is a tab right now can be a settings section.

@jancborchardt @nimishavijay for the actual design specs, I'm just brainstorming :)

Required for https://github.com/nextcloud/calendar/issues/3418

**Comment by ChristophWurst** [comment]
Looks great! We can work on this improvement later. 

For the short term we'll then need another approach, the redesigned cards won't help us with the room suggestions. After all the design doesn't even contain the option to book a room or resource. The main driver for moving to a modal was https://github.com/nextcloud/calendar/issues/3418#issuecomment-941092056.

**Comment by nekohayo** [comment]
I simply wanted to say here that I'm very happy to see some design work happening on moving towards a cards/overlay/popover design rather than the existing sidebar, because the biggest annoyance I have with the current calendar UI is that it requires opening and closing the sidebar all the time, which has two issues:
* It's slow. There is noticeable latency to open and load the sidebar, every time.
* It shifts/squeezes the calendar's view, which is very visually annoying. This may sound silly, but I really can't stand it, the UI bouncing around like a compressed sponge makes it "annoying" to interact with.

I am under the impression that you were not redesigning specifically for _this_ but to solve other issues (with rooms etc.), but I'm mentioning the papercuts above so that those considerations are on your radar if they weren't already.

**Comment by jancborchardt** [comment]
Status: This needs a little rework from the design team to make it accessible.

**Comment by nimishavijay** [comment]
Reworked it a bit more. As a first step, the event card can have a "view" and "edit" mode and show some more relevant info like guests.

View mode:
![image](https://github.com/nextcloud/calendar/assets/52440189/9cf4ef3d-a377-48bb-bc56-4de169f8712f)

- [ ] Remove the illustration on the top
- [ ] Move the calendar indication to the top
- [ ] Show shortened more easily readable date time
- [ ] If there are any links in location or description, make them clickable 
- [ ] Show some attendees (limit to 3, show organiser and confirmed attendees up top)
  - [ ] Nice to have: show an overview like `7 invited, 3 confirmed`
- [ ] Editing is not allowed, need to click on an "Edit" button to enable editing (to avoid accidental editing, similar to contacts)
- [ ] "More details" opens sidebar

Edit mode:
![image](https://github.com/nextcloud/calendar/assets/52440189/d78d4e54-ee80-46d1-86eb-c68ab7d14e13)

- [ ] Editing is allowed now, can be saved with a "Save" button
- [ ] Use new internal label input components 
- [ ] Calendar picker can be a button instead of a dropdown, and clicking on it opens an action menu with the different calendars
- [ ] Allow to quickly add attendees also, show the 3 most recently added attendees
- [ ] "More details" opens sidebar


In the sidebar (screenshot only to show structure, not a pixel-perfect mockup as this was done using the devtools 😅 )
![image](https://github.com/nextcloud/calendar/assets/52440189/ff75acc4-c860-4c4a-bcbe-78afe17b890b)

- [ ] Sidebar doesn't have view/edit distinction, always in editing like how it is currently
- [ ] Header part is similar to the event card, no illustration, calendar picker on top as a button, show location and description as part of the header 
- [ ] Use internal label text inputs (not shown in screenshot) and Vue components for checkboxes, multiselect items etc
- [ ] Small adjustments to the attachments section:
  - [ ] Height of each item is 44px
  - [ ] Size of thumbnail is 24px
  - [ ] Thumbnails are aligned with icons and filename is aligned with rest of the text
  - [ ] Attachment filenames should not be in bold
  - [ ] If attachments are there, show "3 attachments" in bold
- [ ] Follow up: small changes to recurrence section (todo)

What do you think? In scope? Any feedback from @nextcloud/designers also welcome :)

**Comment by st3iny** [comment]
Thank you for the detailed mockups :)

I'm starting to implement the improvements and realized that is is not possible to move an event to another calendar. The calendar dot and name is shown at the top of the popover and sidebar editors but I don't see a way to change it.

Did I miss something? Otherwise, I'd appreciate some feedback on how to implement the calendar picker.

cc @nimishavijay

**Comment by nimishavijay** [comment]
> Did I miss something? 

🤔 This is what I mean when I say calendar picker, the calendar that an event is in is changeable right? Currently it looks like this:

![image](https://github.com/nextcloud/calendar/assets/52440189/dc87af96-b6b1-439f-b3fa-6650b53c8edf)

> A button makes sense but were should it be put?

At the top! :) "Personal" is the calendar the event is a part of, and clicking on it opens an action menu with the different calendars that the user has. 

![image](https://github.com/nextcloud/calendar/assets/52440189/6ef5a22f-ffa0-4579-86c4-1c6799256112)

Did I misunderstand your question? :)

**Comment by st3iny** [comment]
Unfortunately, using text fields with internal labels and icons is not deliverable at the moment. There are multiple reasons:

1. The `NcTextArea` component (used for the description) does not support internal icons yet.
2. We are still on nextcloud/vue 7 but the migration is already in progress: https://github.com/nextcloud/calendar/pull/5648

The rest of the feedback can be delivered. We can move the conversion of all input fields to a follow-up PR. It can be done once `NcTextArea` is fixed and we successfully migrated to the required nextcloud/vue version, which should be either 8 or 9.

**PS:** About `NcTextArea`: I implemented the component initially and decided to drop the internal icon feature for the time being. It turned out to be very complex and there was no need back then. I'll start working on it again after implementing the redesign.
